troposphere/static/js/components/projects/resources/instance/details/sections/AllocationSourceSection.react.js

```diff
@@ -69,15 +44,13 @@ export default React.createClass({
 
                 <div style={{ marginBottom: "20px" }}>
                     <SelectMenu
-                    defaultId={ id }
-                      list={ allocationSourceList }
-                    optionName={ /* Callback */ item => item.get('name') }
-                      onSelectChange={ this.onSourceChange }
-                    />
+                        current={ current }
+                        optionName={ item => item.get('name') }
+                        list={ allocationSourceList }
+                        onSelect={ elem => this.setState({ current: elem }) } />
```

Comment:
onSelect returns an element of the list, not an index. This component has the list and the index, it can pass the element to the parent.